core/data_availability.go

```diff
@@ -64,6 +64,13 @@ func IsDataAvailable(chain consensus.ChainHeaderReader, block *types.Block) (err
 		return nil
 	}
 
+	sidecars := block.Sidecars()
+	// there has a very special case that nodes only keep 90000 recent blocks/blobs when it enables --pruneancient flag.
```

Comment:
so if `pruneancient` is set, the blocks older than 90000 will also be deleted? Along with blobs.